src/neuroconv/datainterfaces/ophys/scanimage/scanimageimaginginterface.py

```diff
@@ -86,3 +87,256 @@ def get_metadata(self) -> dict:
+            )
+
+        avaliable_channels = parsed_metadata["channel_names"]
+        if channel_name is None:
+            if len(avaliable_channels) > 1:
+                raise ValueError(
+                    "More than one channel is detected! Please specify which channel you wish to load "
+                    "with the `channel_name` argument. To see which channels are available, use "
+                    "`ScanImageTiffSinglePlaneImagingExtractor.get_available_channels(file_path=...)`"
+                )
+            channel_name = avaliable_channels[0]
+        assert channel_name in avaliable_channels, f"Channel {channel_name} not found in the tiff file."
+
+        super().__init__(
+            folder_path=folder_path,
+            file_pattern=file_pattern,
+            channel_name=channel_name,
+            extract_all_metadata=extract_all_metadata,
+            verbose=verbose,
+        )
+
+    def get_metadata(self) -> dict:
+        metadata = super().get_metadata()
+
+        extracted_session_start_time = datetime.datetime.strptime(
+            self.image_metadata["epoch"], "[%Y %m %d %H %M %S.%f]"
+        )
+        metadata["NWBFile"].update(session_start_time=extracted_session_start_time)
+
+        num_channels = self.parsed_metadata["num_channels"]
+        channel_name = self.source_data["channel_name"]
+
+        ophys_metadata = metadata["Ophys"]
+        two_photon_series_metadata = ophys_metadata["TwoPhotonSeries"][0]
+        imaging_plane_metadata = ophys_metadata["ImagingPlane"][0]
+        optical_channel_metadata = [
+            channel for channel in imaging_plane_metadata["optical_channel"] if channel["name"] == channel_name
+        ]
+        if num_channels > 1:
+            channel_name_without_space = channel_name.replace(" ", "")
+            imaging_plane_name = f"ImagingPlane{channel_name_without_space}"
+            imaging_plane_metadata.update(
+                name=imaging_plane_name,
+                optical_channel=optical_channel_metadata,
+            )
+            two_photon_series_metadata.update(
+                name=f"TwoPhotonSeries{channel_name_without_space}",
+                imaging_plane=imaging_plane_name,
+            )
+
+        if self.image_metadata is not None:
+            extracted_description = json.dumps(self.image_metadata)
+            two_photon_series_metadata.update(description=extracted_description)
+
+        return metadata
+
+
+def get_scanimage_version(scanimage_metadata: dict) -> str:
+    """
+    Determine the version of ScanImage that produced the TIFF file.
+
+    Parameters
+    ----------
+    scanimage_metadata : dict
+        Dictionary of metadata extracted from a TIFF file produced via ScanImage.
+
+    Returns
+    -------
+    version : str
+        The version of ScanImage that produced the TIFF file.
+    """
+    if "SI.VERSION_MAJOR" in scanimage_metadata:
```

Comment:
For the most recent scan image files, the full version should be `f"{SI.VERSION_MAJOR}.{SI.VERSION_MINOR}.{SI.VERSION_UPDATE}"` but before 2020 they just had `f"{SI.VERSION_MAJOR}.{SI.VERSION_MINOR}"` and before 2019 they just had `f"{SI.VERSION_MAJOR}"` and before 2016 it was `f"{state.software.version}"` as you know (see [here](https://www.mbfbioscience.com/product-resources/scanimage-resources/scanimage-version-history) for details).

With all that in mind, let's just rename this function to `get_scanimage_major_version` so it's more clear.